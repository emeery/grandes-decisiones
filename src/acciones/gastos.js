import uuid from 'uuid';

export const crearGasto = ({
    descripcion = '', 
    nota = '',
    monto = 0,
    date = 0
 }) => ({
    type: 'SUMA_GASTO',
    gasto: {
        id: uuid(),
        descripcion,
        nota,
        monto,
        date
    }
});
// 
export const removerGasto = ({ id } = {}) => ({
    type: 'REMUEVE_GASTO',
    id
});
// editar descripcion o monto, 
export const editarGasto = (id, update) => ({
    type: 'EDITA_GASTO',
    id,
    update
});