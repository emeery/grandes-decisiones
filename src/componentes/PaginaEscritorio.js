import React from 'react';
import GastosLista from './GastosLista'
import GastosListaFiltros from './GastosListaFiltros';
import GastosSumario from './GastosSumario';
const PaginaEscritorio = () => (
    <div>
        <GastosSumario/> 
        <GastosListaFiltros/>
        <GastosLista/>
    </div>
); export default PaginaEscritorio;

