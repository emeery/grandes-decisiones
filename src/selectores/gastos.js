import moment from 'moment';
// valida las fechas y el texto /
// seleccionaGasto
export default (gastos, {texto, ordenarPor, startDate, endDate}) => {
    return gastos.filter((gasto) => {
        const creadoEnElMomento = moment(gasto.date);
        const fechaInicioMatch = startDate ? startDate.isSameOrBefore(creadoEnElMomento, 'day') : true;
        const fechaFinMatch = endDate ? endDate.isSameOrAfter(creadoEnElMomento, 'day'): true;
        const textoMatch = gasto.descripcion.toLowerCase().includes(texto);
        return textoMatch && fechaInicioMatch && fechaFinMatch;
    }).sort(( a , b )=> {
        if (ordenarPor === 'date') {
            return a.date < b.date ? 1 : -1;
        } else if (ordenarPor === 'monto') {
            return a.monto < b.monto ? 1 : -1;
        }
    });
};

// The selectors are functions that take the status of Redux 
// as an argument and return some data to pass to the component.

// const fechaInicioMatch = typeof fechaInicio !== 'number' || gasto.fecha >= fechaInicio;
// const fechaFinMatch = typeof fechaFin !== 'number' || gasto.fecha <= fechaFinMatch;
