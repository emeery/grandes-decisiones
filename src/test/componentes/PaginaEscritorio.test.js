import React from 'react';
import {shallow} from 'enzyme';
import gastos from '../accesorios/gastos';
import PaginaEscritorio
from '../../componentes/PaginaEscritorio';
test(' el componente escritorio',()=>{
    const envoltura = shallow(<PaginaEscritorio />
    );
    expect(envoltura).toMatchSnapshot();
});