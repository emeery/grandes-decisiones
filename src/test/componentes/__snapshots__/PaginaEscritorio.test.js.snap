// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` el componente escritorio 1`] = `
<div>
  <Connect(GastosSumario) />
  <Connect(GastosListaFiltros) />
  <Connect(GastosLista) />
</div>
`;
